frontend/packages/@depmap/data-explorer-2/src/components/DataExplorerPage/hooks/usePlotData.ts

```diff
@@ -79,7 +80,15 @@ export default function usePlotData(plotConfig: DataExplorerPlotConfig | null) {
         setHadError(true);
         window.console.error(e);
 
-        if (e && typeof e === "object" && "message" in e) {
+        // Distinguish Error from ErrorTypeError
+        if (e instanceof ErrorTypeError) {
+          if (e.errorType === "LARGE_DATASET_READ") {
+            // we happen to want to keep the default message but here we name it explicitly for visibility and maybe change it in the future
+            setErrorMessage(e.message);
```

Comment:
I think for the most part `message` should be for debugging and it should be up to the frontend to determine how it gets displayed to users. But I'm totally fine with keeping it this way for now.